source/allocator.c

```diff
@@ -69,34 +74,31 @@ static void s_default_free(struct aws_allocator *allocator, void *ptr) {
 static void *s_default_realloc(struct aws_allocator *allocator, void *ptr, size_t oldsize, size_t newsize) {
     (void)allocator;
     (void)oldsize;
-#if !defined(_WIN32)
-    if (newsize == 0 || !ptr) {
-        free(ptr);
-        return NULL;
-    }
 
+#if !defined(_WIN32)
     if (newsize <= oldsize) {
         return ptr;
     }
 
     /* newsize is > oldsize, need more memory */
     void *new_mem = s_default_malloc(allocator, newsize);
-    if (new_mem) {
-        memcpy(new_mem, ptr, oldsize);
-        s_default_free(allocator, ptr);
-    }
+    AWS_PANIC_OOM(new_mem, "Unhandled OOM encountered in s_default_malloc");
```

Comment:
At this point, both `new_mem` and `ptr` should be different than NULL.

Reply:
```suggestion
    AWS_PANIC_OOM(new_mem && ptr, "Unhandled OOM encountered in s_default_malloc");
```

Reply:
we don't really care about what ptr is at this point though right?

Reply:
We only need to avoid calling `memcpy` with a NULL pointer. So your current solution also works.